core-plugins/src/test/java/io/cdap/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -372,6 +373,51 @@ public void testPathGlobbing() throws Exception {
     Assert.assertTrue(outputValue.contains("CDAP,Platform"));
   }
 
+  @Test
+  public void testSkipHeader() throws Exception {
+    Schema schema = Schema.recordOf("user",
+                                    Schema.Field.of("body", Schema.of(Schema.Type.STRING)),
+                                    Schema.Field.of("offset", Schema.of(Schema.Type.LONG)));
+    Set<StructuredRecord> expected = ImmutableSet.of(
+      StructuredRecord.builder(schema).set("body", "123").set("offset", 4L).build(),
+      StructuredRecord.builder(schema).set("body", "456").set("offset", 8L).build());
+    Assert.assertEquals(expected, testSkipHeader("text", "", schema));;
+
+    schema = Schema.recordOf("user",
+                             Schema.Field.of("val1", Schema.of(Schema.Type.INT)),
+                             Schema.Field.of("val2", Schema.of(Schema.Type.INT)),
+                             Schema.Field.of("val3", Schema.of(Schema.Type.INT)));
+    expected = ImmutableSet.of(
+      StructuredRecord.builder(schema).set("val1", 1).set("val2", 2).set("val3", 3).build(),
+      StructuredRecord.builder(schema).set("val1", 4).set("val2", 5).set("val3", 6).build());
+    Assert.assertEquals(expected, testSkipHeader("csv", ",", schema));
+    Assert.assertEquals(expected, testSkipHeader("tsv", "\t", schema));
+    Assert.assertEquals(expected, testSkipHeader("delimited", " ", schema));
+  }
+
+  private Set<StructuredRecord> testSkipHeader(String format, String delimeter, Schema schema) throws Exception {
+    File inputFile = temporaryFolder.newFile();
+
+    try (Writer writer = new FileWriter(inputFile)) {
```

Comment:
better to write just one file in the caller and pass in the File as an argument, especially since the caller is now checking the output.

Reply:
the content in the file is different for each delimiter, so we cannot use one single file for testing